### PR TITLE
Import/Export Code

### PR DESCRIPTION
**Pull request by tfpk**
Added import/export code. 

Yet TODO:
- [x] Wording/Icon Changes
- [X] Code Refactoring? (mainly just reordering things)
- [X] File Uploads
- [X] Code for closing the modals could be better (especially when uploading)

### CODE REVIEW

**Comment by tfpk** [comment]
Downloads:
image

Uploads:
image

**Comment by tfpk** [comment]
New install window, with file upload button:
image

**Comment by cyphar** [comment]
Since you're going to be busy with organising murder, do you mind if I carry this one?

**Comment by tfpk** [comment]
Sure I'm happy for you to, but I was just waiting on feedback -- the only thing left should be wording changes and I'm more than happy for someone else to do those if you'd prefer.

Could you let me know what else you're doing so I can can keep abreast of the changes (or tag me when they're ready so I can review them).

**Comment by cyphar** [comment]
Sure. The two main changes (aside from this one) are to give system errors a different skin of some kind and to make the timestamps of messages actually tell you which day the message was sent. I just merged a few minor changes (#45) which includes local storage of basic settings (is the panel open? what is your name? is text-to-speech enabled?).

**Comment by cyphar** [comment]
Here's my quick feedback after playing with it for a little bit:
* Upload and download seem like reasonable names.
  - But the modal titles need to be updated.
* The import modal doesn't appear to use the same monospace font as the export modal (at least on my Linux laptop -- your screenshots appear to use the same font?).
* I think the "exists in import" state should look a bit nicer than having the red-highlighted text "(exists in import)". Maybe the row should be tinted red or something like that (or maybe have another table which says "exists in both" -- or maybe even just auto-uncheck the box). 
* There should be "select all" and "deselect all" buttons for both import tables.
* The import tables feel a little bit cramped, but I'm not sure if there's really a way to make it any prettier.
* It seems like the download/upload buttons are a bit weirdly spaced in relation to the icons (a bit cramped)? Also maybe the icons should be standard download and upload icons?
* Now that there is `--accent-3` when rebasing make sure the button colours are the same for "cancel import" and "reset room" (I don't mind which colour you pick -- actually I haven't checked whether they're the same colour...).

I'm also not completely sure about making the upload buttons say "install" but it's probably okay for now.

**Comment by cyphar** [comment]
But yeah, you will be very busy with murder tomorrow and Sunday so I will take this one off your plate. Thanks so much for working on this though, it makes my life a lot simpler. :relieved:

**Comment by cyphar** [comment]
I've also noticed that bot auto-saving appears to not work with this PR. I will fix that too.

**Comment by cyphar** [comment]
I made a fair number of minor changes, one of the big ones is that the input fields now disappear after you click "import" and the way that identical bots are handled is different. I think this is okay enough to be merged (assuming you think it looks reasonable @tfpk) and we can work on any additional cleanups later.

2021-01-02-234821_1055x652_scrot

**Comment by cyphar** [comment]
(I will add select all and deselect all in some unspecified future time if I decide we actually need it.)

File: client/index.html [hunk 1]
```diff
@@ -317,24 +329,24 @@ <h5>Already Installed Bots</h5>
...
-                  <td><a :href="b.url"><code>{{ b.url }}</code></a></td>
+                  <td>{{ bot.name }}</td>
+                  <td><code>{{ bot.responds_to }}</code></td>
+                  <td><a :href="bot.url"><code>{{ bot.url }}</code></a></td>
                 </tr>
               </tbody>
             </table>
 
-            <h5>Importing Bots</h5>
+            <h5>Imported Bots</h5>
```
Comment (tfpk) [inline client/index.html]: This implies the bot has already been imported, hence why I think I used "Importing". Maybe "New Bots" or "Bots to Import"?

**Review by tfpk — COMMENTED** [review]
Ah yeah I realised this was crappy but didn't have the wear with all to fix it...

Thanks :)

**Comment by tfpk** [comment]
LGTM, nice work, and thanks for getting this over the line.

File: client/index.html (hunk 1 above)
Comment (cyphar) [reply client/index.html]: New bots is probably better.

Comment (cyphar) [reply client/index.html]: Done.

**Comment by cyphar** [comment]
:shrug: I'll merge it now YOLO.

**Pull request merged**